packages/amplify_core/lib/src/types/storage/upload_data_options.dart

```diff
@@ -15,21 +17,26 @@ class StorageUploadDataOptions extends StorageOperationOptions
   /// {@macro amplify_core.storage.upload_data_options}
   const StorageUploadDataOptions({
     super.accessLevel,
+    this.metadata,
     this.pluginOptions,
   });
 
+  /// The metadata attached to the object to be uploaded.
+  final Map<String, String>? metadata;
```

Comment:
Generally would recommend Maps always be non-nullable unless there is a specific reason to be able to distinguish an empty map from a null map.

```suggestion
  const StorageUploadDataOptions({
    super.accessLevel,
    this.metadata = const {},
    this.pluginOptions,
  });

  /// The metadata attached to the object to be uploaded.
  final Map<String, String> metadata;
```